users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -329,40 +482,172 @@ def run_scf_audio_perturbation():
+                    **returnn_args["extra_args"],
+                },
+                **returnn_args,
+            },
+            feature_args=feature_args,
+            lr_args=lr_args,
+            report_args=report_args,
+        )
 
-        Returns:
-            tuple: A tuple containing the key string and the report string.
-        """
+    nn_args, report_args_collection = get_nn_args_baseline(
+        nn_base_args=nn_base_args,
+        num_epochs=450,
+        evaluation_epochs=[350, 400, 450],
+        prefix="conformer_",
+    )
 
-        key_string = ""
-        report_dict = {}
+    returnn_root = CloneGitRepositoryJob(
+        "https://github.com/rwth-i6/returnn",
+        commit="c4d36d06f6465e82a50d400d114259e07b8b0709",
+    ).out_repository
+    returnn_root.hash_overwrite = "returnn_conv_padding"
+    report, _ = run_nn_args(
+        nn_args,
+        report_args_collection,
+        dev_corpora,
+        returnn_root=returnn_root,
+        recog_args={"epochs": [350, 400, 450, "best"]},
+    )
+    return report
 
-        for key, value in args.items():
 
-            if key in ["speed", "tempo", "preemphasis", "non_linearity"]:
-                key_component = f"{key}_{value['prob']}_{value['minimum']}_{value['maximum']}"
-                key_string += key_component
-                report_dict[key] = f"{value['prob']}_{value['minimum']}_{value['maximum']}"
-            elif key == "codecs":
-                codecs_str = "_".join([f"{codec['encoding']}_{codec['prob']}" for codec in value])
-                key_string += f"{key}_{codecs_str}_"
-                report_dict[key] = codecs_str
-            else:
-                raise ValueError(f"Unknown argument name: {key}")
+def run_scf_audio_perturbation_from_checkpoint():
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/ctc/feat/"
+
+    _, nn_system = run_scf_baseline()
+
+    (
+        returnn_datasets,
+        rasr_loss_corpus_path,
+        rasr_loss_corpus_segments,
+        rasr_loss_lexicon_path,
+        dev_corpora,
+    ) = get_datasets(use_multi_proc_dataset=True, pre_process=CodeWrapper("audio_perturb_runner.run"))
+    returnn_args = {
+        "batch_size": 5000,
+        "rasr_binary_path": RASR_BINARY_PATH,
+        "rasr_loss_corpus_path": rasr_loss_corpus_path,
+        "rasr_loss_corpus_segments": rasr_loss_corpus_segments,
+        "rasr_loss_lexicon_path": rasr_loss_lexicon_path,
+        "datasets": returnn_datasets,
+        "conformer_type": "wei",
+        "audio_perturbation": True,
+        "use_multi_proc_dataset": True,
+        "specaug_old": {"max_feature": 15},
+        "extra_args": {
+            "accum_grad_multiple_step": 2,
+            "watch_memory": True,
+            "conv_pad_seq_len_to_power": 1.5,
+        },
+    }
+
+    # usually the lr args would need to be adapted to fit with the checkpoint,
+    # but experiments showed that restarting the scheduling is better
```

Comment:
You also need a baseline for this now without perturbation. Otherwise you cannot separate the effects of perturbation and LR reset.

Reply:
good idea. I will use the same checkpoint and just start again.

Reply:
I suspect that the lr reset will not improve the baseline. It probably only helps the network with the perturbation at the start but it is better to check.